### ISSUE TITLE
[Feature Request] Hotkey for swapping between Memory Pak and Rumble Pak

### ISSUE THREAD

**Issue by dezraj** [project64/project64#2143]
Currently there seem not to be a hotkey for swapping between Memory Pak and Rumble Pak mode in the default input plugin. This function would be useful when playing in full-screen games which require both accessories to be swapped once in a while.

I don't know if it would be complicated, but I suppose it would make emulation more user-friendly if such an option was added.

Thank you so much for your time and work in this great emulator.

**Comment by DerekTurtleRoe** [comment]
Agreed, this seems like a nice quality of life improvement! 👍

**Comment by Squall-Leonhart** [comment]
i disagree, you had to physically remove this on a  console so you should have to put in some work to do it on an emulator to.

**Comment by flagrama** [comment]
There were devices that were both and had a switch.

**Comment by Squall-Leonhart** [comment]
I know, they tended to die and corrupt the saves :D

**Comment by dezraj** [comment]
> i disagree, you had to physically remove this on a console so you should have to put in some work to do it on an emulator to.

I understand your point. But I think it has also been a tradition of emulators to improve the original hardware experience and make it more user-friendly, whether through graphics improvement (up-scaling and texture replacement), or enabling the use of save states and different controllers etc. I suppose this hotkey swap feature for those accessories would approach such a tradition too.

**Comment by project64** [comment]
at some stage I plan to go over all the hot key implementation, at the moment the emulator handles the hotkeys, but it should go through the input plugin, allowing you to map hotkeys to controller buttons.

Plugins should be able to register hotkeys as well, so there could be a hot key in the input plugin to do this, or mute/unmute in audio, etc. 

trying to hack something in like this in the current infrastructure I personally have no interest in, and the restructuring of hot keys I have wanted for a long time, but it has been lower in my priority list.

**Comment by dezraj** [comment]
> at some stage I plan to go over all the hot key implementation, at the moment the emulator handles the hotkeys, but it should go through the input plugin, allowing you to map hotkeys to controller buttons.
> 
> Plugins should be able to register hotkeys as well, so there could be a hot key in the input plugin to do this, or mute/unmute in audio, etc.
> 
> trying to hack something in like this in the current infrastructure I personally have no interest in, and the restructuring of hot keys I have wanted for a long time, but it has been lower in my priority list.

Thank you so much for your work zilmar. I appreciate that you have this feature request considered somehow.

**Comment by V7MoVknqxavTR** [comment]
Please.
